move and revise AWS constraints (#846)
aws/karpenter: website/content/en/pre-docs/provisioner-crd.md

  # These fields vary per cloud provider, see your cloud provider specific documentation
  provider: {}
```

## spec.requirements

Kubernetes defines the following [Well-Known Labels](https://kubernetes.io/docs/reference/labels-annotations-taints/), and cloud providers (e.g., AWS) implement them. They are defined at the "spec.requirements" section of the Provisioner API. 

> These well known labels may be specified at the provisioner level, or in a workload definition (e.g., nodeSelector on a pod.spec). If the two conflict, the provisioner controls. In other words, a podspec may not override the provisioner with respect to constraints. 

Reviewer: ```suggestion
These well known labels may be specified at the provisioner level, or in a workload definition (e.g., nodeSelector on a pod.spec). Nodes are chosen using the intersection of provisioner and pod requirements. If there is no overlap, nodes will not be launched. In other words, a pod's requirements must be a subset of the provisioner's requirements.
```

> For example, the instance type may be specified using a nodeSelector in a workload definition. If the instance type requested at the workload (i.e., pod) level is not included in the provisioner list, Karpenter will not create a node or schedule the pod. 

Reviewer: ```suggestion
For example, an instance type may be specified using a nodeSelector in a pod spec. If the instance type requested is not included in the provisioner list and the provisioner has instance type requirements, Karpenter will not create a node or schedule the pod. 
```

      operator: In
      values: ["m5.large", "m5.2xlarge"]
```

*Override with workload manifest (e.g., pod)*

```yaml
spec:
  template:
    spec:
      nodeSelector:
        node.kubernetes.io/instance-type: m5.large
```

### Availability Zones

- key: `topology.kubernetes.io/zone`
- value example: `us-east-1c`

☁️ **AWS**

- value list: `aws ec2 describe-availability-zones --region <region-name>`

Karpenter can be configured to create nodes in a particular zone. Note that the Availability Zone `us-east-1a` for your AWS account might not have the same location as `us-east-1a` for another AWS account.

[Learn more about Availability Zone
IDs.](https://docs.aws.amazon.com/ram/latest/userguide/working-with-az-ids.html)

### Architecture

- key: `kubernetes.io/arch`
- values
  - `amd64` (default)
  - `arm64`

Karpenter supports `amd64` nodes, and `arm64` nodes.

### Capacity Type

- key: `karpenter.sh/capacity-type`

☁️ **AWS**

- values
  - `spot` (default)
  - `on-demand` 

Karpenter supports specifying capacity type, which is analogous to [EC2 purchase options](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-purchasing-options.html).

> ## spec.provider

Reviewer: I like how you've wired this part up.
